include/aspect/material_model/interface.h

```diff
@@ -217,6 +217,16 @@ namespace aspect
                           const bool use_strain_rates = true);
 
       /**
+       * Copy Constructor.
+       */
+      MaterialModelInputs(const MaterialModelInputs<dim> &material);
+
+      /**
+       * Assignment operator.
+       */
+      MaterialModelInputs &operator=(const MaterialModelInputs &material);
```

Comment:
I'm not actually a fan of this. These objects are big, and copying them is expensive. I'd rather just outlaw it (as it was before, implicitly by just not providing this constructor).